crypto/evp/evp_enc.c

```diff
@@ -600,8 +600,7 @@ int EVP_EncryptUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl,
                       const unsigned char *in, int inl)
 {
     int ret;
-    size_t soutl;
-    int blocksize;
+    size_t soutl, blocksize, inl_ = (size_t)inl;
```

Comment:
The assumption seems to be that `SIZE_MAX > INT_MAX + blocksize`. While I can't imagine a platform and algorithm where that wouldn't be the case, I don't think its guaranteed by the standards?? We do have an existing sanity check to confirm that at least `SIZE_MAX >= INT_MAX`:

https://github.com/openssl/openssl/blob/a20c9b6c13afb71e1dd03bf122673e3093d6c437/test/sanitytest.c#L113-L116

Don't forget of course that `blocksize` comes from a (possibly 3rd party) provider. All reasonable values of `blocksize` are small.....but.....

Reply:
int is guaranteed to be signed and size_t is guaranteed to be unsigned, so SIZE_MAX should always be substantially greater than INT_MAX. The minimum permitted value for INT_MAX (C89) is 32767, so we're safe so long as block sizes are below around 32k, and even then that only applies to systems with 16-bit int.

Moreover, blocksize comes from `ctx->cipher->block_size` which is an int. So in the worst case blocksize=INT_MAX and inl=INT_MAX. Since SIZE_MAX must be greater than 2*INT_MAX (because they are required to be unsigned and signed respectively), there can't be an issue here.

Reply:
> Since SIZE_MAX must be greater than 2*INT_MAX

This is not guaranteed by the standard IIUC. For all reasonable platforms it is probably true. If we are relying on that then we should update the sanitytest to confirm it.

Reply:
SIZE_MAX is guaranteed to be at least 65535 by the standard; INT_MAX is guaranteed to be at least 32767. So 2*32767=65534. int can't be wider than size_t because that's ruled out by the sanity test above.

Nonetheless, may as well amend the sanity test.